### ISSUE TITLE
How I was able to install the Matebook X Pro Hackintosh

### ISSUE THREAD

**Issue by Zendarq** [gnodipac886/MatebookXPro-hackintosh#69]
After several days of trying and research here are the steps that I needed to do in order to get it to work.  I hope this helps anyone that tries to attempt this and saves a bit of time.

- Please [donate ](https://github.com/gnodipac886/MatebookXPro-hackintosh#help-a-broke-student-out)to the creator of this code, its impressive.

**Setup the Matebook X Pro to work with MacOS**
- [Edimax AC1200 Wifi USB Adapter ($20.00)](https://www.amazon.com/gp/product/B01MY7PL10/ref=ppx_yo_dt_b_asin_title_o03_s00?ie=UTF8&psc=1)  FYI the right device drivers are [here](https://github.com/chris1111/Wireless-USB-Adapter).  Don't download it from the actual Edimax website, it doesn't work.
- [Samsung 500 TB portable SSD ($90)](https://www.amazon.com/Samsung-T5-Portable-SSD-MU-PA500B/dp/B073GZBT36/ref=sr_1_3?keywords=samsung+ssd+portable+usb+c&qid=1578832487&s=electronics&sr=1-3)
- [USB Hub Multiport Adapter ](https://www.amazon.com/HooToo-Premium-Adapter-Charging-Chromebook/dp/B01K7C53K2/ref=sxin_3_osp18-8a3d542b_cov?ascsubtag=8a3d542b-ad67-4c87-b571-08d3d234a387&creativeASIN=B01K7C53K2&cv_ct_cx=usb+c+hub&cv_ct_id=amzn1.osp.8a3d542b-ad67-4c87-b571-08d3d234a387&cv_ct_pg=search&cv_ct_wn=osp-search&keywords=usb+c+hub&linkCode=oas&pd_rd_i=B01K7C53K2&pd_rd_r=dcee6758-4f61-495e-962a-3615c35ee65b&pd_rd_w=TE99I&pd_rd_wg=qs16T&pf_rd_p=eb3e5cda-5ec9-4d94-919d-310a5d641b8b&pf_rd_r=PHQ6BRAS0E32HAV5Q4XR&qid=1578832396&s=electronics&tag=thewire06oa-20)- not the cheapest but highly rated ($36)
- Any USB mouse
- [32 GB USB Drive ($10)](https://www.amazon.com/Samsung-MUF-32AB-AM-Plus-32GB/dp/B07D7SX9NS/ref=sr_1_8?keywords=32+GB+USB+drive&qid=1578833036&sr=8-8)
- [Velcro squares ($2)](https://www.amazon.com/gp/product/B001682EOK/ref=ppx_yo_dt_b_asin_title_o00_s00?ie=UTF8&psc=1)
- [Short USBC to USBC 3.1 Cable ($8)](https://www.amazon.com/gp/product/B01LONPUM4/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&psc=1)
- Setup the the Matebook X Pro Firmware to allow it to be be booted via BIOS F2 (update firmware and then Disable Secure Boot)

**Get a proper bootable USB Catalina MacOS with Clover installed on it.**
This is the part that I really struggled on.  How can I get a proper bootable MacOS USB Catalina with Clover?  I tried the instructions from the main page and it kinda worked but it would only setup the "diagnostic tools" instead of the full blown OS.  So I went and borrowed my kid's MacBook Air and created one.  I used the following Mark With Tech [guide](https://markwithtech.com/macos-catalina.html) and it worked spectacularly.  Here is the [YouTube video](https://youtu.be/ekp8w6pel84).
- Format the USB Drive using Disk Utility but you need to do "Show All Devices" to get the real drive so that you can create a Mac OS Extended but with a GUID Partition Map Scheme.  If you don't do that, its a non starter because it won't be bootable.
- Install the installation media by running "
sudo /Applications/Install\ macOS\ Catalina.app/Contents/Resources/createinstallmedia --nointeraction --downloadassets --volume /Volumes/USB/"  with /Volumes/USB being the name of the USB drive.
- Now you need to install Clover but clover is not in the link in the guide but [here](https://github.com/Dids/clover-builder/releases). Grab the latest one and in the assets its the one that is the pkg. (e.g. [Clover_v.2.5r_r5103.pkg](https://github.com/Dids/clover-builder/releases/download/v2.5k_r5103/Clover_v2.5k_r5103.pkg))
- When you run this it will ask to install Clover but this is **SUPER** important.  You need to select the RIGHT drive to install it on.  If you don't you screw up your Windows EFI instead!!! [Don't do it](https://youtu.be/ekp8w6pel84?t=109).   Make sure it is the the USB installation.
- If the step above was done properly, you have an EFI directory partition mounted on your mac. Now there is a section that you need to install the Ktext.  Ignore all that. Take the stuff from this repo's USB-EFI directory and copy it to the EFI directory and replace its contents!
- Copy the rest of the repo into a directory in the original drive mount and the USB Wifi adapter driver as listed in the link above.

**Installation of Catalina on the Matebook X Pro**
- Plug in to the Matebook, the USB Hub with power, USB Mouse, and the Samsung SSD drive (you will format the SSD drive later) 
- Insert the USB Drive into the Matebook while shutdown.  Restart but hold down the F12 button until you get into the EFI Boot process.  If you don't see the USB Mac OS EFI bootable, you did something wrong.  Select the USB bootable drive
- It will go blank and the Apple logo and progress bar will show up.  I will then get into the disk utility setup and do the process again of formatting the Samsung SSD drive with the GUID partition scheme.
- Once it is done then you will then select the "Install Catalina MacOS" and select the Samsung SSD to be the target of the installation.  This will take about 40 minutes.
- It will reboot, and yes it will boot to windows first for some reason even if you press F12.
- Reboot again and then in Clover you will then select the new SSD drive as your startup OS.
- It will be then looking for a network connection, just bypass all that stuff and set it up later.
- You will finally be able to setup your first user !!!!
- After that you should have that original MacOS Catalina USB drive available with the Wifi Drive pkg already in it.  Read the web page about it but essentially it was just install...then a reboot.
- Windows comes up but then reboot again and press F12 and boot back into your Samsung SSD.
- Ok now you are in business

**Installation of Clover on your Samsung SSD with new Catalina OS**
- Download the Clover Pro and that previous Clover pkg.  This time instead of installing it on the USB you are installing it on the Samsung SSD! So you no longer need the USB drive.
- Again remember to select the Samsung SSD as the clover target.
- If successful again the EFI mount will be available for you.  
- Copy the rest of the files from the root of this repo into that EFI mount.  This will setup the proper ktexts required to configure your Matebook X Pro to be legit working.
- Unmount and remove the USB stick, its no longer needed
- Reboot ... windows.... Reboot...F12... Clover .... Samsung SSD selected !  Boom you are now self aware.

**You can now do a proper clean up**
- Shutdown
- Unplug the USB Hub
- Use the 4 of the velcro squares and mount the Samsung SSD to the back of the Matebook.  Don't put it too low or else it will prevent you from pushing the screen back.
- Use the short USB C cable so it looks neater
- Restart F12.  Super clean setup!

Congrats you are done and its an awesome dream device I always dreamed Apple would have made.

**Comment by kian1991** [comment]
Awesome thank you for the guidance! This will help new users for sure

**Comment by svkonopat** [comment]
Do you mean Clover Configurator when you said Clover Pro? Other than this run-in, this guide was easy to follow and very helpful! Thank you!

**Comment by kmonty** [comment]
This was a great guide! The "createinstallmedia" trick vs. attempting to use gibMacOS made my process of upgrading my existing Mojave machine so much easier. Cheers!

**Comment by jonescamilla** [comment]
Hello, I'm trying to follow your guide after days of failed attempts. My question is which settings in the clover install did you use? Did you use the same settings he ticked in his [YouTube](https://www.youtube.com/watch?v=ekp8w6pel84&feature=youtu.be) video?

**Comment by androidek** [comment]
Hi,
I really like your guide - it is really straight forward! Unfortunately, I run into some troubles. I was following the guide directly. When I was using the createinstallmedia command, I had some extra lines there (downloading assets) https://imgur.com/9XJM7k2 but it seems that the whole proccess was successful.

Then, I installed the clover from the link from this guide, copied all the stuff from USD-EFI to EFI folder (& replaced) & I downloaded the wifi drivers.

Then, when I actually tried booting it from the usb, it all initally seemed succesful - ive managed to get to the macos boot screen, but the progress bar stopped around 60% and didn't want to move (i left it for couple of hours). https://imgur.com/2OwxdUi

Any help?

when i was copying usb-efi to efi folder & then the rest of the directory i pasted it like this (https://imgur.com/sU5LOPq ). Install macOS Catalina partition looked like this (https://imgur.com/J06b1WL ).

**Comment by Zendarq** [comment]
> Hi,
> I really like your guide - it is really straight forward! Unfortunately, I run into some troubles. I was following the guide directly. When I was using the createinstallmedia command, I had some extra lines there (downloading assets) https://imgur.com/9XJM7k2 but it seems that the whole proccess was successful.
> 
> Then, I installed the clover from the link from this guide, copied all the stuff from USD-EFI to EFI folder (& replaced) & I downloaded the wifi drivers.
> 
> Then, when I actually tried booting it from the usb, it all initally seemed succesful - ive managed to get to the macos boot screen, but the progress bar stopped around 60% and didn't want to move (i left it for couple of hours). https://imgur.com/2OwxdUi
> 
> Any help?
> 
> when i was copying usb-efi to efi folder & then the rest of the directory i pasted it like this (https://imgur.com/sU5LOPq ). Install macOS Catalina partition looked like this (https://imgur.com/J06b1WL ).

Could be the Hard drive support?  I never had an issue with that.

**Comment by androidek** [comment]
I have managed to complete the hackintosh installation! My method was the following. I used clover installation method from the og post and then created catalina installer from your method (& Mark With Tech). Thanks a lot anyway! Hackintosh works very closely to perfection!

**Comment by seanfarr788** [comment]
> I have managed to complete the hackintosh installation! My method was the following. I used clover installation method from the og post and then created catalina installer from your method (& Mark With Tech). Thanks a lot anyway! Hackintosh works very closely to perfection!

Congratulations on your install. Would.you however be able to clarify how you were able to do this however? I'm a little confused how you were able to install Catalina without formatting the disk (which thus gets rid of what you did using the "og" you rep)

Many thanks

**Comment by androidek** [comment]
I initially followed method above to create bootable usb (but when I tried to boot it, progress bar stopped about 60%). 
So then, I tried using the og method (Download gibMacOS Scripts, Run gibMacOS.bat, Run MakeInstall.bat (this step formats the disk & installs the bootable macos on pendrive), Copy clover files from this repo onto your USB). However, when I tried installing Catalina (or later Mojave), this installer required internet connection to run (I didn't have ethernet adapter, so I was stuck). Mostly by accident, i tried booting from the usb created from  the og method and then ran bootable macos from the usb created from method above. This worked and didn't require internet connection.

it might be a little bit complicated, but to put it in simple terms: I created bootable USB using Zendarq's method. Then I created bootable usb using gnodipac886's method (this guide helped me a lot https://www.youtube.com/watch?v=NWkn0YfkhSQ) and then booted clover from gnodipac886's USB and ran macos install from Zendarq's USB

**Comment by therealmrsymba** [comment]
This guide was very helpful! Thanks!
